src/reducers/fileTree.tsx

```diff
@@ -191,6 +190,100 @@ export const getRelativePath = ({
   return pathList[newIndex];
 };
 
+const findRelativeMessageUid = (
+  currentPath: string,
+  messageMap: LinterMessageMap,
+  pathList: string[],
+  position: RelativePathPosition,
+): LinterMessage['uid'] | null => {
+  let path = currentPath;
+  for (let i = 0; i < pathList.length; i++) {
+    const nextPath = getRelativePath({
+      currentPath: path,
+      pathList,
+      position,
+    });
+    const nextMessages = messageMap[nextPath];
+    if (nextMessages) {
+      const nextMessagesForPath = getMessagesForPath(nextMessages);
+      if (position === RelativePathPosition.previous) {
+        const lastMessageIndex = nextMessagesForPath.length - 1;
+        if (lastMessageIndex >= 0) {
```

Comment:
This needs a test because I don't see any failures after this change:

```diff
diff --git a/src/reducers/fileTree.tsx b/src/reducers/fileTree.tsx
index c956ffd..e092303 100644
--- a/src/reducers/fileTree.tsx
+++ b/src/reducers/fileTree.tsx
@@ -208,9 +208,7 @@ const findRelativeMessageUid = (
       const nextMessagesForPath = getMessagesForPath(nextMessages);
       if (position === RelativePathPosition.previous) {
         const lastMessageIndex = nextMessagesForPath.length - 1;
-        if (lastMessageIndex >= 0) {
-          return nextMessagesForPath[lastMessageIndex].uid;
-        }
+        return nextMessagesForPath[lastMessageIndex].uid;
       } else if (nextMessagesForPath.length) {
         return nextMessagesForPath[0].uid;
       }
```

Reply:
Looking at this again, I don't think we need this `if` statement at all. I think this was a result of some verbose code I wrote and then refactored. This `if` statement is checking to make sure that what we calculate as the index of the last message for the file (via `nextMessagesForPath.length - 1`) is in fact a valid index, but as long as `nextMessagesForPath.length` is not `0`, it will always be valid, and if `nextMessagesForPath.length` *is* `0` we don't even need to do this calculation. I will make some changes to the code so this `if` statement is no longer needed.